scripts/prepare_for_registration_bulk_creation.py

```diff
@@ -0,0 +1,155 @@
+logging.basicConfig(level=logging.INFO)
+
+
+@celery_app.task(name='scripts.prepare_for_registration_bulk_creation')
+def prepare_for_registration_bulk_creation(parsing_output, dry_run=False):
+
+    if not parsing_output:
+        logger.error('Missing CSV input')
+        return handle_internal_error()
+        # The following are temporarily hard-coded and used for local implementation/testing only. The format of
+        # the `csv_input` has mostly been determined, except for the `csv_parsed` which needs collaboration with
+        # the parser piece. The value of `csv_parsed` is expected to be a serializable object that can be pickled
+        # and un-pickled by celery automatically.
+        # parsing_output = {
+        #     'payload_hash': '07eef7ee-fc69-4e3a-9047-8d0e0a4cb909',
+        #     'provider_id': 'osf',
+        #     'schema_id': '60f74311ed3441000141b265',
+        #     'initiator_id': 't47ka',
+        #     'registrations': [
+        #         {
+        #             'external_id': '27093472',
+        #             'csv_raw': '1997,Ford,F250,Black',
+        #             'csv_parsed': {
+        #                 'year': 1997,
+        #                 'make': 'Ford',
+        #                 'model': 'F250',
+        #                 'color': 'Black',
+        #             },
+        #         },
+        #         {
+        #             'external_id': '98709987',
+        #             'csv_raw': '2000,Volkswagen,Golf,White',  # here goes the raw string of one row
+        #             'csv_parsed': {
+        #                 'year': 2000,
+        #                 'make': 'Volkswagen',
+        #                 'model': 'Golf',
+        #                 'color': 'White',
+        #             },
+        #         },
+        #         {
+        #             'external_id': '64510752',
+        #             'csv_raw': '2011,Subaru,WRX,Blue',  # here goes the raw string of
+        #             'csv_parsed': {
+        #                 'year': 2011,
+        #                 'make': 'Subaru',
+        #                 'model': 'WRX',
+        #                 'color': 'Blue',
+        #             },
+        #         },
+        #     ],
+        # }
+
+    logger.info('Preparing OSF database for registration bulk creation ...')
+    initiator_id = parsing_output.get('initiator_id', None)
+    if not initiator_id:
+        logger.error('Missing initiator id')  # internal error
+        return handle_internal_error()
+    initiator = OSFUser.load(initiator_id)
+    if not initiator:
+        logger.error('Initiator not found: [id={}]'.format(initiator_id))  # internal error
+        return handle_internal_error()
+
+    provider_id = parsing_output.get('provider_id', None)
+    if not provider_id:
+        logger.error('Missing provider id')  # internal error
+        return handle_internal_error()
+    provider = RegistrationProvider.objects.filter(_id=provider_id).first()
```

Comment:
Is is possible for two providers to have the same id?

Reply:
Yes!!! RegistrationProvider inherits from `TypedObjectIDMixin` which doesn't have the unique constraint on `_id`. It depends on if registration provider all have the same type.

```python 
class TypedObjectIDMixin(ObjectIDMixin):
    class Meta:
        abstract = True
        # On subclasses, be sure to add a unique_together constraint on '_id' and 'type'

    _id = models.CharField(max_length=24, default=generate_object_id, db_index=True)
```
It is in `AbstractProvider` that defines `unique_together = ('_id', 'type')`

```python
class AbstractProvider(TypedModel, TypedObjectIDMixin, ReviewProviderMixin, DirtyFieldsMixin, BaseModel):
    class Meta:
        unique_together = ('_id', 'type')
        permissions = REVIEW_PERMISSIONS
unique_together = ('_id', 'type')
```

Reply:
At the database level, preprints provider and registration provider shares the same table `osf_abstractprovider`. However, can I assume that `RegistrationProvider.objects.filter(...)` already grantees the type?